cmd/utils/fdlimit_windows.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package utils
+
+// #include <stdio.h>
+import "C"
+
+// raiseFdLimit tries to maximize the file descriptor allowance of this process
+// to the maximum hard-limit allowed by the OS.
+func raiseFdLimit(max uint64) error {
+	if max > 2048 {
+		max = 2048 // windows limits
+	}
+	C._setmaxstdio(C.int(max))
```

Comment:
I'm not sure that setmaxstdio will help us. The [docs](https://msdn.microsoft.com/en-us/library/aa298575(v=vs.60).aspx) say that it raises the limit for handles at "stdio" level up to 2048.
(*opened and accessed by means of the fopen, fgetc, fputc, and so forth family of functions*).
The "lowio" level is capped at 2048 by default.

However, package os uses the [`CreateFile`](https://msdn.microsoft.com/en-us/library/windows/desktop/aa363858(v=vs.85).aspx) function from winapi to open files.
I don't know which limit applies there.